src/module.cc

```diff
@@ -0,0 +1,388 @@
+  if (nodereport_events & NR_FATALERROR) {
+    TriggerNodeReport(Isolate::GetCurrent(), kFatalError, message, location, NULL);
+  }
+  fflush(stderr);
+  if (nodereport_core) {
+    raise(SIGABRT); // core dump requested (default)
+  } else {
+    exit(1); // user specified that no core dump is wanted, just exit
+  }
+}
+
+bool OnUncaughtException(v8::Isolate* isolate) {
+   // Trigger NodeReport if required
+  if (nodereport_events & NR_EXCEPTION) {
+    TriggerNodeReport(isolate, kException, "exception", "OnUncaughtException (nodereport/src/module.cc)", NULL);
+  } 
+  if (nodereport_core) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+#ifndef _WIN32
+static void SignalDumpInterruptCallback(Isolate *isolate, void *data) {
+  if (report_signal != 0) {
+    if (nodereport_verbose) {
+      fprintf(stdout,"nodereport: SignalDumpInterruptCallback handling signal\n");
+    }
+    if (nodereport_events & NR_SIGNAL) {
+      if (nodereport_verbose) {
+        fprintf(stdout,"nodereport: SignalDumpInterruptCallback triggering NodeReport\n");
+      }
+      TriggerNodeReport(isolate, kSignal_JS,
+                        node::signo_string(report_signal),
+                        "node::SignalDumpInterruptCallback()", NULL);
+    }
+    report_signal = 0;
+  }
+}
+static void SignalDumpAsyncCallback(uv_async_t* handle) {
+  if (report_signal != 0) {
+    if (nodereport_verbose) {
+      fprintf(stdout,"nodereport: SignalDumpAsyncCallback handling signal\n");
+    }
+    if (nodereport_events & NR_SIGNAL) {
+      if (nodereport_verbose) {
+        fprintf(stdout,"nodereport: SignalDumpAsyncCallback triggering NodeReport\n");
+      }
+      TriggerNodeReport(Isolate::GetCurrent(), kSignal_UV,
+                        node::signo_string(report_signal),
+                        "node::SignalDumpAsyncCallback()", NULL);
+    }
+    report_signal = 0;
+  }
+}
+
+/*******************************************************************************
+ * Utility functions for signal handling support (platforms except Windows)
+ *  - RegisterSignalHandler() - register a raw OS signal handler
+ *  - SignalDump() - implementation of raw OS signal handler
+ *  - StartWatchdogThread() - create a watchdog thread
+ *  - ReportSignalThreadMain() - implementation of watchdog thread
+ *  - SetupSignalHandler() - initialisation of signal handlers and threads
+ ******************************************************************************/
+ // Utility function to register an OS signal handler
+static void RegisterSignalHandler(int signal, void (*handler)(int signal)) {
+  struct sigaction sa;
+  memset(&sa, 0, sizeof(sa));
+  sa.sa_handler = handler;
+  sigfillset(&sa.sa_mask);  // mask all signals while in the handler
+  sigaction(signal, &sa, &saved_sa);
+}
+
+// Utility function to restore an OS signal handler to its previous setting
+static void RestoreSignalHandler(int signal) {
+  sigaction(signal, &saved_sa, nullptr);
+}
```

Comment:
As an aside: maybe `s/signal/signo/` because that a) doesn't shadow the libc function of the same name, and b) is consistent with the function below.